src/runtime/security/security.thrift

```diff
@@ -0,0 +1,46 @@
+include "../../dsn.thrift"
+
+namespace cpp dsn.security
+
+// negotiation process:
+//
+//                       client                              server
+//                          | ---   SASL_LIST_MECHANISMS    --> |
+//                          | <-- SASL_LIST_MECHANISMS_RESP --- |
+//                          | --    SASL_SELECT_MECHANISMS  --> |
+//                          | <-- SASL_SELECT_MECHANISMS_OK --- |
+//                          |                                   |
+//                          | ---      SASL_INITIATE        --> |
+//                          |                                   |
+//                          | <--      SASL_CHALLENGE       --- |
+//                          | ---  SASL_CHALLENGE_RESPONSE  --> |
+//                          |                                   |
+//                          |              .....                |
```

Comment:
what is doing in this procedure?

Reply:
There maybe a couple of repeats of `SASL_CHALLENGE` and `SASL_CHALLENGE_RESPONSE`. It's not very clearly to show how many couples will be repeated, because it relys on the implement of security protocol.